Framework/PythonInterface/plugins/algorithms/LoadLamp.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import (absolute_import, division, print_function)
+from mantid.api import FileProperty, WorkspaceProperty, PythonAlgorithm, NumericAxis, AlgorithmFactory, FileAction
+from mantid.kernel import Direction
+from mantid.simpleapi import CreateWorkspace, AddSampleLogMultiple, mtd
+
+import numpy
+import h5py
+
+
+class LoadLamp(PythonAlgorithm):
+    def name(self):
+        return 'LoadLamp'
+
+    def category(self):
+        return 'DataHandling\\Nexus'
+
+    def summary(self):
+        return 'Loads HDF files exported from LAMP program at the ILL'
+
+    def PyInit(self):
+        self.declareProperty(FileProperty(name="InputFile", defaultValue="", action=FileAction.Load, extensions=["hdf"]))
+        self.declareProperty(WorkspaceProperty(name="OutputWorkspace", defaultValue="", direction=Direction.Output))
+
+    def PyExec(self):
+        input_file = self.getProperty("InputFile").value
+        output_ws = self.getPropertyValue("OutputWorkspace")
+
+        with h5py.File(input_file, 'r') as hf:
+            DATA = numpy.array(hf.get('entry1/data1/DATA'), dtype='float')
```

Comment:
Is there any reason for the variable names being in upper case? As far as I can tell, they're not globals? `DATA` is also quite a confusing name - what's the relationship between `DATA`, `Y` and `y_axis`, and could this be made any clearer?

Reply:
The reason I had in mind, is to make one to one correspondence with the entry name in the HDF file. But it is true that they are not globals, so maybe it is more correct to rename them.  `y_axis` is just a mantid object.